src/partial_span_processor/__init__.py

```diff
@@ -113,15 +114,32 @@ def get_heartbeat_attributes(self) -> dict[str, str]:
     return {
       "partial.event": "heartbeat",
       "partial.frequency": str(self.heartbeat_interval_millis) + "ms",
+      "partial.body.type": "json/v1",
     }
 
   def get_log_data(self, span: Span, attributes: dict[str, str]) -> LogData:
-    span_context = Span.get_span_context(span)
+    instrumentation_scope = span.instrumentation_scope if hasattr(span,
+                                                                  "instrumentation_scope") else None
+    span_context = span.get_span_context()
 
     enc_spans = encode_spans([span]).resource_spans
     traces_data = trace_pb2.TracesData()
     traces_data.resource_spans.extend(enc_spans)
-    serialized_traces_data = traces_data.SerializeToString()
+    serialized_traces_data = json_format.MessageToJson(traces_data)
+
+    # FIXME/HACK replace serialized traceId and spanId values as string comparison
```

Comment:
not the happiest solution but built-in json serialization does not serialize trace_id and span_id from span according to partial collector expectations.

Reply:
adding a test for a hack seems silly to me since this is using built-in json serialization.